packages/client/src/Components/Mapping/components/FormChildPerception/index.jsx

```diff
@@ -0,0 +1,70 @@
+import React, { useState } from 'react'
+import _ from 'lodash'
+
+const Perception = ({ store, onStoreUpdate, callbackFunction }) => {
+  const [currentForce] = useState(store.currentForce)
+  const [currentMarkerName] = useState(store.currentMarkerName)
+  const [forcePerception, setForcePerception] = useState(store.perception.force)
+  const [typePerception, setTypePerception] = useState(store.perception.type)
+  const [allForces] = useState(store.allForces)
+  const [allPlatforms] = useState(store.allPlatforms)
+
+  // A copy of the store to capture the updates
+  const newStore = store
+
+  const handleSubmit = (e) => {
+    e.preventDefault()
+    callbackFunction(newStore)
+  }
+
+  const handleRadioChange = ({ target }) => {
+    setForcePerception(target.value)
+
+    newStore.perception.force = target.value
+    // save data in helper class to not lose it after popup recreate
+    onStoreUpdate(newStore)
+  }
+
+  const handleSelectChange = ({ target }) => {
```

Comment:
Wouldn't `handlePlatformTypeChange` be closer to expressing business logic in the variable name?

Reply:
completely agree. Done